src/components/common/settings/sidebar.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { ChevronLeft } from "lucide-react";
+import Link from "next/link";
+import { usePathname, useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+
+import { Settings_Sidebar_Links } from "~/lib/constant";
+import { cn } from "~/lib/utils";
+
+const SettingsSidebar = () => {
+  const [activeLink, setActiveLink] = useState("");
+  const router = useRouter();
+  const pathname = usePathname();
+  // remove the / from the pathname
+  const currentPath = pathname?.replace("/settings/", "");
+
+  useEffect(() => {
+    setActiveLink(currentPath);
+  }, [currentPath]);
```

Comment:
looks like active link is just the currentPath so we can actually use that directly and avoid having to sync it

Reply:
reason for this

```
`  // remove the /settings/ from the pathname
  const currentPath = pathname?.replace("/settings/", "");

  useEffect(() => {
    setActiveLink(currentPath);
  }, [currentPath]);`
```